pkg/render/intrusion_detection.go

```diff
@@ -1136,14 +1175,255 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"k8s-app": ADAPIObjectName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: ADAPIObjectPortName,
+					Port: adAPIPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: adAPIPort,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIDeployment() *appsv1.Deployment {
+	adAPIStorageVolumePath := "/storage"
+	adAPIStorageVolumeName := "volume-storage"
+	envVars := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "info"},
+		{Name: "STORAGE_PATH", Value: adAPIStorageVolumePath},
+		{Name: "TLS_KEY", Value: c.cfg.ADAPIServerCertSecret.VolumeMountKeyFilePath()},
+		{Name: "TLS_CERT", Value: c.cfg.ADAPIServerCertSecret.VolumeMountCertificateFilePath()},
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+			Labels: map[string]string{
+				"k8s-app": ADAPIObjectName,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &adAPIReplicas,
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": ADAPIObjectName}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ADAPIObjectName,
+					Namespace: IntrusionDetectionNamespace,
+					Labels: map[string]string{
+						"k8s-app": ADAPIObjectName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: ADAPIObjectName,
+					// setting tolerations, nodeselector as the node affinity setting respecting
+					// the control plane tagged node as the centeralized node for anomaly
+					// detection storage
+					Tolerations:      c.cfg.Installation.ControlPlaneTolerations,
+					NodeSelector:     c.cfg.Installation.ControlPlaneNodeSelector,
+					ImagePullSecrets: secret.GetReferenceList(c.cfg.PullSecrets),
+					Volumes: []corev1.Volume{
+						c.cfg.TrustedCertBundle.Volume(),
+						c.cfg.ADAPIServerCertSecret.Volume(),
+						{
+							Name: adAPIStorageVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  ADAPIObjectName,
+							Image: c.adAPIImage,
+							Env:   envVars,
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/health",
+										Port:   intstr.FromInt(adAPIPort),
+										Scheme: corev1.URISchemeHTTPS,
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       10,
+								FailureThreshold:    5,
```

Comment:
Not sure how there are handled in the other places of the operator. Is there any precedent for these numbers?

Reply:
going to remove them and use k8s default https://github.com/tigera/operator/pull/1931/commits/243589f6f789db15732483f8ba163a1b349e89bc